src/plugins/intel_gpu/src/kernel_selector/kernel_base.cpp

```diff
@@ -90,34 +90,54 @@ JitConstants KernelBase::MakeBaseParamsJitConstants(const base_params& params, b
     jit.Merge(MakeActivationJitConstants(params.activations, unitType));
 
     if (add_tensor_definitions) {
-        size_t dyn_tensor_idx = 0;
-
+        size_t shape_info_offset = 0;
         for (size_t i = 0; i < params.inputs.size(); i++) {
-            jit.AddConstant(MakeJitConstant("INPUT" + toCodeString(i), params.inputs[i], dyn_tensor_idx));
+            jit.AddConstant(MakeJitConstant("INPUT" + toCodeString(i), params.inputs[i], shape_info_offset));
+            for (auto j : params.inputs[i].GetDims()) {
+                if (j.pad.is_dynamic) {
+                    shape_info_offset += 2;
+                }
+            }
             if (params.inputs[i].is_dynamic())
-                dyn_tensor_idx++;
+                shape_info_offset += 6;
```

Comment:
Could you please add some constants for these numbers (6, 2)?

Reply:
I will do.

Reply:
Done